presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/ResourceGroupNameTemplate.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.execution.resourceGroups;
+
+import com.facebook.presto.SessionRepresentation;
+import com.fasterxml.jackson.annotation.JsonCreator;
+
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+public class ResourceGroupNameTemplate
+{
+    private static final Pattern USER_PATTERN = Pattern.compile(Pattern.quote("${USER}"));
+    private static final Pattern SOURCE_PATTERN = Pattern.compile(Pattern.quote("${SOURCE}"));
+    private static final Pattern PARAMETER_PATTERN = Pattern.compile("\\$\\{(.*?)\\}");
+
+    private final String name;
+
+    @JsonCreator
+    public ResourceGroupNameTemplate(String name)
+    {
+        this.name = requireNonNull(name, "name is null");
+        checkArgument(!name.isEmpty(), "Resource group name is empty");
+        checkArgument(ResourceGroupId.REGEXP.matcher(name).matches(), "Invalid resource group name. '%s' contains a '.'", name);
+        Matcher matcher = PARAMETER_PATTERN.matcher(name);
+        while (matcher.find()) {
+            String group = matcher.group(1);
+            checkArgument(group.equals("USER") || group.equals("SOURCE"), "Unsupported template parameter: ${%s}", group);
+        }
+    }
+
+    public String expandTemplate(SessionRepresentation session)
+    {
+        String expanded = USER_PATTERN.matcher(name).replaceAll(session.getUser());
+        return SOURCE_PATTERN.matcher(expanded).replaceAll(session.getSource().orElse(""));
```

Comment:
If you remember if the name contains either of these it would be faster... I don't think it really matters